src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -2362,6 +2366,129 @@ JSC_DEFINE_CUSTOM_GETTER(jsSqlStatementGetParamCount, (JSGlobalObject * lexicalG
     RELEASE_AND_RETURN(scope, JSC::JSValue::encode(JSC::jsNumber(sqlite3_bind_parameter_count(castedThis->stmt))));
 }
 
+JSC_DEFINE_CUSTOM_GETTER(jsSqlStatementGetColumnTypes, (JSGlobalObject * lexicalGlobalObject, JSC::EncodedJSValue thisValue, PropertyName attributeName))
+{
+    auto& vm = JSC::getVM(lexicalGlobalObject);
+    JSSQLStatement* castedThis = jsDynamicCast<JSSQLStatement*>(JSValue::decode(thisValue));
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    CHECK_THIS
+    CHECK_PREPARED
+
+    // Ensure the statement has been executed at least once
+    if (!castedThis->hasExecuted) {
+        throwException(lexicalGlobalObject, scope, createError(lexicalGlobalObject, "Statement must be executed before accessing columnTypes"_s));
+        return { };
+    }
+
+    int count = sqlite3_column_count(castedThis->stmt);
+    JSC::JSArray* array = JSC::constructEmptyArray(lexicalGlobalObject, nullptr, count);
+    
+    // We need to reset and step the statement to get fresh types,
+    // but only do this for read-only statements to avoid side effects
+    bool isReadOnly = sqlite3_stmt_readonly(castedThis->stmt) != 0;
+
+    if (! isReadOnly) {
+      // For non-read-only statements, throw an error since column types don't make sense
+      throwException(lexicalGlobalObject, scope, createError(lexicalGlobalObject, "columnTypes is not available for non-read-only statements (INSERT, UPDATE, DELETE, etc.)"_s));
+      return { };
+    }
+
+    // Reset the statement (safe for read-only statements)
+    int resetStatus = sqlite3_reset(castedThis->stmt);
+    if (resetStatus != SQLITE_OK) {
+        throwException(lexicalGlobalObject, scope, createSQLiteError(lexicalGlobalObject, castedThis->version_db->db));
+        return { };
+    }
+
+    // Step once to get to the first row (safe for read-only statements)
+    int stepStatus = sqlite3_step(castedThis->stmt);
+
+    // If we got a row, get types from it
+    if (stepStatus == SQLITE_ROW) {
+        for (int i = 0; i < count; i++) {
+            JSC::JSValue typeValue;
+
+            // Get the actual column type from the current row
+            int columnType = sqlite3_column_type(castedThis->stmt, i);
+
+            switch (columnType) {
+            case SQLITE_INTEGER:
+                typeValue = JSC::jsNontrivialString(vm, "INTEGER"_s);
```

Comment:
Let's use atom strings so that string comparisons on these strings are faster

```suggestion
                typeValue = JSC::jsAtomString(vm, makeAtomString("INTEGER"_s));
```

Reply:
Thanks @Jarred-Sumner, and apologies for wasting your time with such trivial fixes